src/templates/dso_api/dynamic_api/api.html

```diff
@@ -63,8 +67,458 @@
+    } else if (!isPaginated) {
+      // If paginated but pagination has not yet been set (for page 1)
+      setPageLinks(page);
+    }
+  }
+
+  function exceptionHandler(e) {
+      console.log("Kan data niet ophalen.");
+  }
+
+  function parseData(data) {
+    let contentElement = document.getElementById("response-content");
+
+    // We need to mark coordinates fields so we can make them collapsible.
+    let markCoordinates = (key, value) => {
+        if (key == "coordinates"){
+            return "__coordinatestring__" + JSON.stringify(value, null, 4);
+        }
+        return value;
+    }
+    
+    // Activate next link
+    if(data["_links"] && data["_links"]['next']) {
+      document.getElementById("page-links").getElementsByTagName("li").item(6).classList.remove("disabled");
+    }
+
+    // Create highlighted html string and set content
+    jsonString = JSON.stringify(data, markCoordinates,4);
+    contentElement.innerHTML = syntaxHighlight(jsonString);
+  }
+
+  function syntaxHighlight(jsonString) {
+      jsonString = jsonString.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
+      return jsonString.replace(/("(\\u[a-zA-Z0-9]{4}|\\[^u]|[^\\"])*"(\s*:)?|\b(true|false|null)\b|-?\d+(?:\.\d*)?(?:[eE][+\-]?\d+)?)/g, function (match) {
+          let cls = 'number';
+          if (/^"https?:\/\//.test(match)) {
+              // Create a link for strings containing an url.
+              return '<a href=' + match + ' rel="nofollow"><span class="link">' + match +'</span></a>'
+          }
+          if (/^"__coordinatestring__/.test(match)) {
+              // Cut the marker and enclosing quotes from the coordinates string and unescape newlines.
+              match = match.substring(21,match.length-1).replace(/\\n/g, "\n");
+
+              // Grab the head and tail for the summary and remove whitespace.
+              let head = match.substring(0,100).replace(/\n| /g, "");
+              let tail = match.substring(match.length-100).replace(/\n| /g, "");
+
+              // Create an element that can be expanded with a click.
+              let result = '<span class="collapsible collapsed" onClick="this.classList.toggle(\'collapsed\')">';
+              result += '<span class="collapsible_summary">'+syntaxHighlight(head)+'...'+syntaxHighlight(tail)+'</span>';
+              result += '<span class="collapsible_content">'+syntaxHighlight(match)+'</span></span>';
+              return result;
+          }
+          if (/^"/.test(match)) {
+              if (/:$/.test(match)) {
+                  cls = 'key';
+              } else {
+                  cls = 'string';
+              }
+          } else if (/true|false/.test(match)) {
+              cls = 'bool';
+          } else if (/null/.test(match)) {
+              cls = 'null';
+          }
+          return '<span class="' + cls + '">' + match + '</span>';
```

Comment:
template strings?

```suggestion
          return `<span class="${cls}">${match}</span>`;
```

en HTML escaping?

Reply:
Ja top dat is veel mooier. Eerste regel van syntaxHighlight doet de html escaping al.